run/grpc-ping/main.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Sample relay acts as an intermediary to the ping service.
+package main
+
+import (
+	"log"
+	"net"
+	"os"
+
+	"google.golang.org/grpc"
+
+	pb "github.com/GoogleCloudPlatform/golang-samples/run/grpc-ping/pkg/api/v1"
+)
+
+func main() {
+	log.Printf("grpc-ping: starting server...")
+
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = "8080"
+		log.Printf("Defaulting to port %s", port)
+	}
+
+	listener, err := net.Listen("tcp", ":"+port)
+	if err != nil {
+		log.Fatalf("net.Listen: %v", err)
+	}
+
+	grpcServer := grpc.NewServer()
+	pb.RegisterPingServiceServer(grpcServer, &pingService{})
+	if err = grpcServer.Serve(listener); err != nil {
+		log.Fatal(err)
+	}
+}
+
+// [END run_grpc_server]
+
+// Conn holds an open connection to the ping service.
```

Comment:
Why outside the sample?

Reply:
The scope of the how-to guide is not how to use gRPC, but the specific points of intersection with Cloud Run. The current expectation is this means showing how to start a gRPC server listening on $PORT and making service-to-service gRPC requests.

I've excluded showing the environment variables as they seem outside scope of the snippets the docs need.